arbitrum-client/src/client/contract_interaction.rs

```diff
@@ -0,0 +1,199 @@
+//! Defines `ArbitrumClient` helpers that allow for interacting with the
+//! darkpool contract
+
+use circuit_types::{merkle::MerkleRoot, wallet::Nullifier, PlonkProof};
+use constants::{Scalar, ScalarField};
+
+use crate::{
+    errors::ArbitrumClientError,
+    helpers::{deserialize_retdata, serialize_calldata},
+    serde_def_types::SerdeScalarField,
+    types::{
+        MatchPayload, Proof, ValidMatchSettleStatement, ValidWalletCreateStatement,
+        ValidWalletUpdateStatement,
+    },
+};
+
+use super::ArbitrumClient;
+
+// TODO: Replace `renegade_contracts_common::types::*` with relayer statement
+// types once they're adapted to Plonk
+
+impl ArbitrumClient {
+    // -----------
+    // | GETTERS |
+    // -----------
+
+    /// Get the current Merkle root in the contract
+    pub async fn get_merkle_root(&self) -> Result<ScalarField, ArbitrumClientError> {
+        let merkle_root_bytes = self
+            .darkpool_contract
+            .get_root()
+            .call()
+            .await
+            .map_err(|e| ArbitrumClientError::ContractInteraction(e.to_string()))?;
+
+        Ok(deserialize_retdata::<SerdeScalarField>(&merkle_root_bytes)?.0)
```

Comment:
nit: Add a comment here (maybe inline) for the `.0` just like we do when we pass magic numbers as arguments, it's unclear to the reader what the zeroth field is